src/main/java/com/azure/cosmos/examples/analyticalcontainercrud/async/AnalyticalContainerCRUDQuickstartAsync.java

```diff
@@ -0,0 +1,214 @@
+        // them
+        CosmosContainerResponse containerResponse = database.createContainerIfNotExists(containerProperties).block();
+        CosmosAsyncContainer container = database.getContainer(containerResponse.getProperties().getId());
+
+        logger.info("Done.");
+    }
+
+    // Update container throughput
+    private void updateContainerThroughput() throws Exception {
+        logger.info("Update throughput for container " + containerName + ".");
+
+        // Specify new throughput value
+        ThroughputProperties throughputProperties = ThroughputProperties.createManualThroughput(800);
+        container.replaceThroughput(throughputProperties).block();
+
+        logger.info("Done.");
+    }
+
+    // Container read
+    private void readContainerById() throws Exception {
+        logger.info("Read container " + containerName + " by ID.");
+
+        // Get container id
+        container = database.getContainer(containerName);
+
+        logger.info("container id: "+container.getId());
+    }
+
+    // Container read all
+    private void readAllContainers() throws Exception {
+        logger.info("Read all containers in database " + databaseName + ".");
+
+        // Read all containers in the account
+        CosmosPagedFlux<CosmosContainerProperties> containers = database.readAllContainers();
+
+        // Print
+        String msg = "Listing containers in database:\n";
+        containers.byPage(100).flatMap(readAllContainersResponse -> {
+            logger.info("read " +
+                    readAllContainersResponse.getResults().size() + " containers(s)"
+                    + " with request charge of " + readAllContainersResponse.getRequestCharge());
+
+            for (CosmosContainerProperties response : readAllContainersResponse.getResults()) {
+                logger.info("container id: " + response.getId());
+                // Got a page of query result with
+            }
+            return Flux.empty();
+        }).blockLast();
+        // for(CosmosContainerProperties containerProps : containers) {
+        // msg += String.format("-Container ID: %s\n",containerProps.getId());
+        // }
+        logger.info(msg + "\n");
+
+        logger.info("Done.");
+    }
+
+    // Container delete
+    private void deleteAContainer() throws Exception {
+        logger.info("Delete container " + containerName + " by ID.");
+
+        // Delete container
+        CosmosContainerResponse containerResp = database.getContainer(containerName)
+                .delete().doOnError(throwable -> {
+                    logger.warn("Delete container {} failed ", containerName, throwable);                    
+                }).doOnSuccess(response -> {
+                    logger.info("Delete container {} succeeded: {}", containerName);
+                }).block();
+        logger.info("Status code for container delete: {}", containerResp.getStatusCode());
```

Comment:
I like the .doOnError/.doOnSuccess usage.
Line #182 seems unnecessary as the result already got logged in .doOnError/.doOnSuccess block.